modules/core/exported/callbacks.go

```diff
@@ -0,0 +1,19 @@
+package exported
+
+type CallbackPacketDataI interface {
```

Comment:
There was a user on Discord asking about the flow of token transfer + ICA call (i.e. on acknowledgement of a token transfer, execute ICA call to send a transaction to the interchain account). It seems like the user was not using a smart contract, so I suggested to use the message router to send a `MsgSendTx` to the interchain account sub-controller. After discussing with @womensrights, for such flow we will probably need also an ADR-8 interface so that when processing the token transfer acknowledgement, the necessary information to send a TX to an interchain account is provided (e.g connection ID, owner address, maybe interchain account packet data?). If a new interface is required (maybe is good to have separate interfaces for different use cases), should we rename this interface to something like `ContractCallbackPacketDataI` to make it specific to the use case of smart contracts?

Reply:
They should still be able to use the interfaces defined here. They would have to make their own custom middleware to do the go logic though.